### ISSUE TITLE
Self-edges generated by some plugins

### ISSUE THREAD

**Issue by jvolkman** [stackb/rules_proto#205]
Thanks for v2! Love the new direction with Gazelle.

I'm running into an issue with self-edges being generated by some of the plugins. This seems to happen when a `.proto` file imports another file within the same package. I put together a small example at https://github.com/jvolkman/rules-proto-self-edge-bug.

In this example, the `proto_cc_library` target [depends on itself](https://github.com/jvolkman/rules-proto-self-edge-bug/blob/main/proto/BUILD.bazel#L27), and the `proto_go_library` lists itself in [`embed`](https://github.com/jvolkman/rules-proto-self-edge-bug/blob/main/proto/BUILD.bazel#L64). This is the result of running `bazel run //:gazelle`.

Does this look familiar? Maybe I've got something misconfigured.

**Comment by jvolkman** [comment]
I spent a bit more time on this and found that a file is generated correctly the first time (i.e., no BUILD.bazel file exists yet), but the second time `gazelle` is run, the self edges are added.

Example, using the repo linked in the original message:
```shell
$ rm proto/BUILD.bazel 

$ bazel run //:gazelle
INFO: Analyzed target //:gazelle (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle-runner.bash
  bazel-bin/gazelle
INFO: Elapsed time: 0.047s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action

$ bazel query //proto:test_go_proto --output=build
go_library(
  name = "test_go_proto",
  visibility = ["//visibility:public"],
  generator_name = "test_go_proto",
  generator_function = "proto_go_library",
  generator_location = "proto-self-edge-bug/proto/BUILD.bazel:55:17",
  srcs = ["//proto:foo.pb.go", "//proto:test.pb.go"],
  deps = ["@org_golang_google_protobuf//reflect/protoreflect:protoreflect", "@org_golang_google_protobuf//runtime/protoimpl:protoimpl"],
  importpath = "github.com/cedarai/tms/proto/test",
)

$ bazel run //:gazelle                            
INFO: Analyzed target //:gazelle (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle-runner.bash
  bazel-bin/gazelle
INFO: Elapsed time: 0.048s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action

$ bazel query //proto:test_go_proto --output=build
go_library(
  name = "test_go_proto",
  visibility = ["//visibility:public"],
  generator_name = "test_go_proto",
  generator_function = "proto_go_library",
  generator_location = "proto-self-edge-bug/proto/BUILD.bazel:58:17",
  srcs = ["//proto:foo.pb.go", "//proto:test.pb.go"],
  deps = ["@org_golang_google_protobuf//reflect/protoreflect:protoreflect", "@org_golang_google_protobuf//runtime/protoimpl:protoimpl"],
  importpath = "github.com/cedarai/tms/proto/test",
  embed = ["//proto:test_go_proto"],
)
```

Running a second time resulted in the `embed` line being added to the target.

**Comment by jvolkman** [comment]
Okay, so `depsresolver.go` does try to filter out self imports [here](https://github.com/stackb/rules_proto/blob/master/pkg/protoc/depsresolver.go#L109), but there's a mismatch in my case between labels stored in the `RuleIndex` and `from`. `from` is prefixed with the local repo name, whereas entries in the index are not. The discrepancy causes `IsSelfImport` to always return false.

#208 strips the local repo name. Maybe there's a better solution.

**Comment by jvolkman** [comment]
Closed #208 because it's unnecessary (and harmful) with Gazelle builds following [this commit](https://github.com/bazelbuild/bazel-gazelle/commit/9beec7e51be76b83a8e613b36801b028608f3d1e). 0.24.0 - the current Gazelle release - doesn't include the commit, so self-import filtering is broken. However, after updating to Gazelle's HEAD, everything works as expected.

**Comment by jvolkman** [comment]
Closing this since things seem to work as expected going forward.
